src/components/FormModal/FormModal.js

```diff
@@ -92,27 +98,100 @@ export default function FormModal({ onCancelAdding, id, name }) {
   }
 
   function handleMorningClicked(name) {
-    const newMorningChecked = weekRoutine.days.map((day) => {
-      if (day.name === name) {
-        day.morning = !day.morning;
-        return day;
-      } else {
-        return day;
-      }
+    const i = getIndexForWeekday(name);
+    const checkedMornings = routineData.filter(
+      (product) =>
+        product.days[i].name === name && product.days[i].morning === true
+    );
+
+    const routineIDs = checkedMornings.map((product) => product.id);
+    const intersection = routineIDs.filter((id) => conflicts.includes(id));
+
+    const conflictName = intersection.map((id) => {
+      const conflictProduct = getProductById(id, products);
+      return conflictProduct.name;
     });
```

Comment:
You could chain this. Save some line of code.
```suggestion
    const conflictName = routineData.filter( 
    (product) => product.days[i].name === name && product.days[i].morning === true)
    .map((product) => product.id)
    .filter((id) => conflicts.includes(id))
    .map((id) => {
      const conflictProduct = getProductById(id, products);
      return conflictProduct.name;
    })
```

Reply:
I'll chain the first bit that gets me to the intersection definition and then i will do a separate map for the conflictName, so i can still use the intersection array as a condition for below :)

Reply:
Very good idea.